include/macroni/Common/CodeGenDriverSetup.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "vast/CodeGen/CodeGenDriver.hpp"
+#include "vast/Util/Common.hpp"
+#include <clang/AST/ASTContext.h>
+#include <memory>
+
+namespace macroni {
+// Creates a vast::cg::driver type with all MLIR and VAST dialects preloaded,
+// along with the given dialect and metagenerator, from the given AST. The
+// driver does not have any visitors loaded into it, so callers will have to set
+// up the visitor stack themselves.
+template <typename dialect_t, typename metagen_t>
+std::unique_ptr<vast::cg::driver>
+generate_codegen_driver(vast::acontext_t &actx) {
+  // Load MLIR and VAST dialects into registry.
+  mlir::DialectRegistry registry;
+  mlir::registerAllDialects(registry);
+  vast::registerAllDialects(registry);
```

Comment:
^ separate helper, I may have seen this already.

Reply:
Not sure where else you saw it in Macroni?

Reply:
I can't use `vast::cg::mk_mcontext()` because that only registers mlir's and vast's dialect and then loads all available dialects, whereas I need to also load my Macroni dialect before loading all available dialects.

Reply:
I mean as long as this is our only usage in the project it is fine this way. The comment was more aimed at me noticing the same thing (copy & paste) in some previous PR and was not sure if this is it or something different. On itself there's no issue with this.